src/bin/lib/pg/snprintf.c

```diff
@@ -341,14 +337,37 @@ static void leading_pad(int zpad, int signvalue, int *padlen,
 						PrintfTarget *target);
 static void trailing_pad(int padlen, PrintfTarget *target);
 
+/*
+ * If strchrnul exists (it's a glibc-ism), it's a good bit faster than the
+ * equivalent manual loop.  If it doesn't exist, provide a replacement.
+ *
+ * Note: glibc declares this as returning "char *", but that would require
+ * casting away const internally, so we don't follow that detail.
+ */
+#ifndef HAVE_STRCHRNUL
+
 static inline const char *
-pg_strchrnul(const char *s, int c)
+strchrnul(const char *s, int c)
```

Comment:
I see it's already changed back, still I think it should have a comment. Same for the `include` changes that we made. Otherwise next release we will do the same thing again.

Reply:
Good point, fixed now.